src/patchCreateProgram.ts

```diff
@@ -0,0 +1,62 @@
+import * as ts from 'typescript'
+import {PluginCreator} from './PluginCreator'
+
+export interface BaseHost {
+    createProgram(
+        rootNames: ReadonlyArray<string>,
+        options: ts.CompilerOptions,
+        host?: ts.CompilerHost,
+        oldProgram?: ts.Program
+    ): ts.Program
+    versionMajorMinor: string
+}
+
+export function patchCreateProgram<Host extends BaseHost>(
+    tsm: Host
```

Comment:
Why is not just `tsm: typeof ts`

Reply:
Conflicts with tsserverlibrary and typescriptServices declarations

Reply:
Ok, so can fix it to 
```
export type BaseHost = Pick<typeof ts, 'createProgram' | 'versionMajorMinor'>
```
to be more confident that if ts team will change `createProgram` interface we will get an error